app/helpers/blacklight_helper.rb

```diff
@@ -10,7 +10,16 @@ def render_search_bar
   end
 
   def render_expert_help(specialists)
-    specialists.present? && specialists.items.first.hits > 50000 ? (render partial: 'catalog/expert_help', locals: {subject: specialists.items.first.value}) : (render partial: 'catalog/ask')
+    if specialists.present? && specialists.items.first.hits > 50000
+      subject = specialists.items.first.value
+      specialist_data = Rails.cache.fetch(:subject_specialist_data, expires_in: 24.hours) do
```

Comment:
What does Rails.cache do if the "refresh" block throws an error when it attempts to refresh (like if the site is down or 404 or something?

Reply:
Related: wondering what happens if the format of the "live" source changes ... i.e., not a 404 or whatever, but some refactoring of the way the source page presents its info. I could see an argument for not accounting for that in code, figuring it's something that would fail in a visible way and could be fixed reactively. An more defensive alternative might be to do some validation in the refresh block (assuming there's some way that Rails.cache can continue using stale data in the event that the refresh block raises an exception) ...

Reply:
Interesting...perhaps we need some error handling here to be sure that this cache fetch doesn't end up replacing a "well-formed" Hash of specialist data if the block here returns nil, `{}` or raises an exception, etc. I'll see what I can do!

Reply:
Yeah that's exactly the lines along which I was thinking. It may actually already do that (depending on how Rails.cache.fetch works), but it'd be good to no what to expect in any case. Also, kind of an edge case, but because the containers that get deployed in docker are stateless (i.e., new container every time service scaled up or down), worst case would be if you happen to scale up when source url is down you could have _no_ specialist data, stale or otherwise. I wonder if still having a fallback file serialized in the repo would make sense ... would also answer the case of offline development -- which I guess doesn't matter so much if you can tether to smartphone data, but still ... :-)

Reply:
I've introduced some sanity checks for the cached data and for attempts to pull data from Drupal which should ensure we are working with legit data when it matters (see `seems_legit?`). I configured Faraday to retry the request to Drupal up to 2 times to handle intermittent issues. If all else fails, it _should_ just render the generic template for the expert help area.